Map

MeuhV1 opened the pull request.

dguenms reviewed (approved):
Recently I have also noticed that if you place a landmark that isn't associated with a player (i.e. with the world builder tool) and then save from WB you trigger an exception and the resulting file is in a corrupted state. I think it is because it is trying to create a player instance to write a comment into the file. It seems related to this, can you also look into it?

MeuhV1 commented:
I'll look into it.

MeuhV1 commented:
Got it to work. :D

dguenms commented on Assets/Python/pyWB/CvWBDesc.py:
```diff
@@ -1809,7 +1809,10 @@ def write(self, f, sign):
 		f.write("\tplotX=%d\n" %(sign.getPlot().getX(),))
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		if sign.getPlayerType() == -1:
```
Are you sure this works? Where is iPlayer defined?

MeuhV1 commented:
You're right about iPlayer not being defined. I found what caused the error and got it to work. After a successful test, I removed some testing code. Apperantly I removed 1 line too many. But this does fix it. (I tested it just before the push to be sure)

dguenms commented on Assets/Python/pyWB/CvWBDesc.py:
```diff
@@ -1809,7 +1809,11 @@ def write(self, f, sign):
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		iPlayer = sign.getPlayerType()
+		if sign.getPlayerType() == -1:
```
This can now also be `iPlayer`

dguenms commented on Assets/Python/pyWB/CvWBDesc.py:
```diff
@@ -1809,7 +1809,11 @@ def write(self, f, sign):
 		f.write("\tplotX=%d\n" %(sign.getPlot().getX(),))
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		iPlayer = sign.getPlayerType()
```
This looks weird on Github, as if there was an additional linebreak. Are there trailing whitespaces?

MeuhV1 replied:
The is no additional linebreak. There are also no trailing whitespaces. I guess it is just Github being weird.

I posted the new code in the post in the link below, so you can view it without Github peculiarities.
https://forums.civfanatics.com/threads/modmodding-q-a-thread.516755/page-94#post-15139548

EDIT:
Reuploaded to include iPlayer == -1, as you suggested.

Commit:
Platybuilder can remove landmarks
Changed the map size values of Rhyes.h
Can WB save with landmarks
